File: genius-blog-main/src/main/java/io/github/geniusay/controller/BlogController.java
package io.github.geniusay.controller;

import io.github.common.web.Result;
import io.github.geniusay.delegate.BlogDelegate;
import io.github.geniusay.pojo.dto.BlogDTO;
import io.github.id.snowflake.SnowflakeRegisterException;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import javax.annotation.Resource;
import javax.validation.Valid;

@RestController
@RequestMapping("/app/v1/blog")
public class BlogController {

    @Resource
    private BlogDelegate delegate;

    @PostMapping(value = "/addBlog")
    public Result addBlog(@RequestBody BlogDTO blogDTO) throws SnowflakeRegisterException {
        return delegate.addBlog(blogDTO);
    }

    @PostMapping(value = "/uploadImg")
    public Result uploadImg(MultipartFile img) throws SnowflakeRegisterException {
        return delegate.uploadImg(img);
    }

    @GetMapping(value = "/blogPageList")
    public Result blogPageList(@RequestParam("page") int page,@RequestParam("limit") int limit) {
        return delegate.blogPageList(page,limit);
    }

    @GetMapping(value = "/oneBlog")
    public Result oneBlog(@RequestParam("blog_id") Long blogId) {
        return delegate.queryById(blogId);
    }
}

File: genius-blog-front/src/api/BlogController.js
import request from '@/utils/request';


export function uploadImg(img ) {
     return request({
        url: '/app/v1/blog/uploadImg/',
        method: 'post',
        data: {
            img:img,
        }
     });
}
export function addBlog(blogDTO ) {
     return request({
        url: '/app/v1/blog/addBlog/',
        method: 'post',
        data: {
            blogDTO:blogDTO,
        }
     });
}
export function oneBlog(blogId ) {
     return request({
        url: '/app/v1/blog/oneBlog/',
        method: 'get',
        params: {
            blogId:blogId,
        }
     });
}
export function blogPageList(page, limit ) {
     return request({
        url: '/app/v1/blog/blogPageList/',
        method: 'get',
        params: {
            page:page,
            limit:limit,
        }
     });
}

